iquip/apps/dataviewer.py

```diff
@@ -238,3 +240,169 @@ def __init__(self, parent: Optional[QWidget] = None):
+    
+    Attributes:
+        seriesLabel: Label showing the name of the current data series (dataset).
+        numberOfSamplesBox: Spin box showing the total number of samples.
+        thresholdBox: Spin box for setting the threshold for state discrimination.
+        buttonGroup: Data type selection radio button group.
+        valueBoxes: Dict of spin boxes for each data type.
+
+    Signals:
+        dataTypeChanged(dataType): Current data type is changed to dataType.
+        thresholdChanged(threshold): Current threshold value is changed to threshold.
+    """
+
+    class DataType(enum.IntEnum):
+        """Type of each data point.
+        
+        Each item is used as its index in the button group.
+        """
+        TOTAL = 0
+        AVERAGE = 1
+        P1 = 2
+
+    dataTypeChanged = pyqtSignal(DataType)
+    thresholdChanged = pyqtSignal(int)
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        layout = QGridLayout(self)
+        # first column (general information)
+        self.seriesLabel = QLabel("", self)
+        self.numberOfSamplesBox = QSpinBox(self)
+        self.numberOfSamplesBox.setMaximum(2**31 - 1)
+        self.numberOfSamplesBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.numberOfSamplesBox.setReadOnly(True)
+        self.numberOfSamplesBox.setFrame(False)
+        self.thresholdBox = QSpinBox(self)
+        self.thresholdBox.setMaximum(2**31 - 1)
+        self.thresholdBox.setButtonSymbols(QAbstractSpinBox.PlusMinus)
+        self.thresholdBox.setPrefix("> ")
+        seriesLayout = QHBoxLayout()
+        seriesLayout.addWidget(QLabel("Series: "))
+        seriesLayout.addWidget(self.seriesLabel)
+        numberOfSamplesLayout = QHBoxLayout()
+        numberOfSamplesLayout.addWidget(QLabel("#Samples: "))
+        numberOfSamplesLayout.addWidget(self.numberOfSamplesBox)
+        thresholdLayout = QHBoxLayout()
+        thresholdLayout.addWidget(QLabel("Threshold: "))
+        thresholdLayout.addWidget(self.thresholdBox)
+        firstColumn = seriesLayout, numberOfSamplesLayout, thresholdLayout
+        for row, item in enumerate(firstColumn):
+            layout.addLayout(item, row, 0)
+        # second column (data type selection)
+        self.buttonGroup = QButtonGroup(self)
+        self.valueBoxes: Dict[DataPointWidget.DataType, QSpinBox] = {}
+        for dataType in DataPointWidget.DataType:
+            button = QRadioButton(dataType.name.capitalize(), self)
+            self.buttonGroup.addButton(button, id=dataType)
+            if dataType is DataPointWidget.DataType.TOTAL:
+                spinbox = QSpinBox(self)
+                spinbox.setMaximum(2**31 - 1)
+            else:
+                spinbox = QDoubleSpinBox(self)
+                if dataType is DataPointWidget.DataType.P1:
+                    spinbox.setMaximum(1)
+                else:
+                    spinbox.setMaximum(np.inf)
```

Comment:
It looks little strange that the maximum of the total is INT_MAX, but that of the average is unbounded..

Reply:
`INT_MAX` is kind of equivalent to _unbounded_ in our case, since it will never reach that value.